sdk/src/main/java/io/opentelemetry/sdk/trace/export/SpanData.java

```diff
@@ -0,0 +1,379 @@
+        .resource(resource)
+        .name(name)
+        .kind(kind)
+        .startTimestamp(startTimestamp)
+        .attributes(
+            Collections.unmodifiableMap(
+                new HashMap<>(Utils.checkNotNull(attributes, "attributes"))))
+        .timedEvents(
+            Collections.unmodifiableList(
+                new ArrayList<>(Utils.checkNotNull(timedEvents, "timedEvents"))))
+        .links(Collections.unmodifiableList(new ArrayList<>(Utils.checkNotNull(links, "links"))))
+        .status(status)
+        .endTimestamp(endTimestamp)
+        .build();
+  }
+
+  /**
+   * Returns the {@code SpanContext} associated with this {@code Span}.
+   *
+   * @return the {@code SpanContext} associated with this {@code Span}.
+   * @since 0.1.0
+   */
+  public abstract SpanContext getContext();
+
+  /**
+   * Returns the parent {@code SpanId}. If the {@code Span} is a root {@code Span}, the SpanId
+   * returned will be invalid..
+   *
+   * @return the parent {@code SpanId} or an invalid SpanId if this is a root {@code Span}.
+   * @since 0.1.0
+   */
+  public abstract SpanId getParentSpanId();
+
+  /**
+   * Returns the resource of this {@code Span}.
+   *
+   * @return the resource of this {@code Span}.
+   * @since 0.1.0
+   */
+  public abstract Resource getResource();
+
+  /**
+   * Returns the name of this {@code Span}.
+   *
+   * @return the name of this {@code Span}.
+   * @since 0.1.0
+   */
+  public abstract String getName();
+
+  /**
+   * Returns the kind of this {@code Span}.
+   *
+   * @return the kind of this {@code Span}.
+   * @since 0.1.0
+   */
+  public abstract Kind getKind();
+
+  /**
+   * Returns the start {@code Timestamp} of this {@code Span}.
+   *
+   * @return the start {@code Timestamp} of this {@code Span}.
+   * @since 0.1.0
+   */
+  public abstract Timestamp getStartTimestamp();
+
+  /**
+   * Returns the attributes recorded for this {@code Span}.
+   *
+   * @return the attributes recorded for this {@code Span}.
+   * @since 0.1.0
+   */
+  public abstract Map<String, AttributeValue> getAttributes();
+
+  /**
+   * Returns the timed events recorded for this {@code Span}.
+   *
+   * @return the timed events recorded for this {@code Span}.
+   * @since 0.1.0
+   */
+  public abstract List<TimedEvent> getTimedEvents();
+
+  /**
+   * Returns links recorded for this {@code Span}.
+   *
+   * @return links recorded for this {@code Span}.
+   * @since 0.1.0
+   */
+  public abstract List<io.opentelemetry.trace.Link> getLinks();
+
+  /**
+   * Returns the {@code Status}.
+   *
+   * @return the {@code Status}.
+   * @since 0.1.0
+   */
+  public abstract Status getStatus();
+
+  /**
+   * Returns the end {@code Timestamp}.
+   *
+   * @return the end {@code Timestamp}.
+   * @since 0.1.0
+   */
+  public abstract Timestamp getEndTimestamp();
+
+  /**
+   * The trace id for this span, retrieved from the associated SpanContext.
+   *
+   * @return the TraceId
+   */
+  public TraceId getTraceId() {
```

Comment:
Consider to either remove the getContext or the individual things. May be better to just keep all the getters here and remove the getSpanContext.

Reply:
how about making the getContext() method package-access, and expose the pieces as getters that delegate? Then, we don't need to add 4 new fields to the `SpanData` class itself to replace the context.

Reply:
I went ahead and removed the context and put the 4 fields in its place.  Feels like the thing has grown a bit, but it's fine.  👍